### ISSUE TITLE
[nodejs] error: Failed while trying to navigate

### ISSUE THREAD

**Issue by 2gn** [nix-community/dream2nix#503]
I have an electron project and I tried to package it with dream2nix following the official tutorial. And I stuck at running `nix build .#default`

(store hash is modified)

```shell
user@computer> nix build .                                                                                 ~/Developer/project
warning: Git tree '/Users/user/Developer/project' is dirty
error: Failed while trying to navigate to node_modules/7zip-bin from /nix/store/0r09dsrfgnmdkwf0c6qr7f84hzx32a5pl-source/
(use '--show-trace' to show detailed location information)
```

The directory structure of this project looks like this:
```
build
dist
flake.lock
flake.nix
main.js
main.js.map
main.ts
node_modules
package.json
pages
preload.js
projects.toml
renderer.js
yarn.lock
```

And the source copied to /nix/store has these

```shell
user@computer> ls /nix/store/0r09dsrfgnmdkwf0c6qr7f84hzx32a5pl-source/                                      ~/Developer/project
flake.lock    package.json    preload.js      renderer.js
flake.nix       main.ts         pages           projects.toml
```

Here's my flake.nix
```nix
{
  inputs = {
    dream2nix.url = "github:nix-community/dream2nix";
  };
  outputs = {
    self,
    dream2nix,
  } @ inp:
    (dream2nix.lib.makeFlakeOutputs {
      systems = [
        "x86_64-linux"
        "x86_64-darwin"
      ];
      config.projectRoot = ./.;
      source = ./.;
      projects = ./projects.toml;
    });
}
```

projects.toml seems to contain whole list of packages, thus it's too big to paste here

**Comment by DavHau** [comment]
Seems like some packages have been detected accidentally in your node_modules directory.
This can be caused by an invalid lock file.
Try the following:
- delete node_modules
- delete your lock file (yarn.lock or package-lock.json)
- re-create lock file
- re-generate prpjects.toml

If it still doesn't work, please share the contents of you projects.toml

**Comment by 2gn** [comment]
I tried the following:
```shell
user@computer> rm -rf node_modules yarn.lock projects.toml
user@computer> nix-shell -p nodejs yarn
[nix-shell:~/Developer/project]$ yarn install #generates yarn.lock
```
and the commands written in the documentation, but I still get the same error.

> If it still doesn't work, please share the contents of you projects.toml

https://gist.github.com/2gn/86f543f7e422fa77cdddc722f1acbd18

**Comment by 2gn** [comment]
I noticed that `nix flake show` fails as well.

```terminal
user@computer> nix flake show                                                                              ~/Developer/project
warning: Git tree '/Users/user/Developer/project' is dirty
git+file:///Users/user/Developer/project
├───apps
│   └───x86_64-darwin
│       └───detect-projects: app
├───devShells
│   └───x86_64-darwin
error: Failed while trying to navigate to node_modules/@swc/core-darwin-x64/package.json from /nix/store/k1f8hjylz8qnif9l43wym0mizany
m2ql-source/
(use '--show-trace' to show detailed location information)
```

**Comment by zmrocze** [comment]
@2gn Imo the issue title is very poor, because `nix build .` is such a fundamental command that any issue in the nix ecosystem can be renamed to "`nix build .` fails".
Some suggestion: 
[nodejs] Error: "Failed while trying to navigate to node_modules/7zip-bin"

**Comment by 2gn** [comment]
> [nodejs] Error: "Failed while trying to navigate to node_modules/7zip-bin"

I think it’s not 

> @2gn Imo the issue title is very poor, because `nix build .` is such a fundamental command that any issue in the nix ecosystem can be renamed to "`nix build .` fails".
> Some suggestion:
> [nodejs] Error: "Failed while trying to navigate to node_modules/7zip-bin"

I think that the issue is not only for 7zip. It was evaluated first because it comes first alphabetically. 
I do agree `nix build . causes an error ` is a bit too ambiguous.

**Comment by zmrocze** [comment]
@2gn if you're still having this issue, check out the fix from the above linked issue (#520). It may work for you and it would confirm the cause.

**Comment by 2gn** [comment]
I’ll try it, thanks. If I could fix my problem using your fix, I’ll close this issue.
